### PR TITLE
Ue5.3

### PR DESCRIPTION
**Pull request by Frickerson**

### CODE REVIEW

File: Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddInputContextMapping.cpp [hunk 1]
```diff
@@ -56,6 +72,113 @@ EDataValidationResult UGBFGameFeatureAction_AddInputContextMapping::IsDataValid(
...
+        for ( auto local_player_iterator = game_instance->GetLocalPlayerIterator(); local_player_iterator; ++local_player_iterator )
+        {
+            RegisterInputMappingContextsForLocalPlayer( *local_player_iterator );
+        }
+    }
+}
+
+void UGBFGameFeatureAction_AddInputContextMapping::RegisterInputMappingContextsForLocalPlayer( ULocalPlayer * local_player )
+{
+    if ( ensure( local_player ) )
```
Comment (TheEmidee) [inline Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddInputContextMapping.cpp]: != nullptr

invert the condition also

File: Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddInputContextMapping.cpp [hunk 2]
```diff
@@ -56,6 +72,113 @@ EDataValidationResult UGBFGameFeatureAction_AddInputContextMapping::IsDataValid(
...
+        for ( auto local_player_iterator = game_instance->GetLocalPlayerIterator(); local_player_iterator; ++local_player_iterator )
+        {
+            UnregisterInputMappingContextsForLocalPlayer( *local_player_iterator );
+        }
+    }
+}
+
+void UGBFGameFeatureAction_AddInputContextMapping::UnregisterInputMappingContextsForLocalPlayer( ULocalPlayer * local_player )
+{
+    if ( ensure( local_player ) )
```
Comment (TheEmidee) [inline Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddInputContextMapping.cpp]: != nullptr + invert condition

File: Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddWidget.cpp [hunk 3]
```diff
@@ -1,6 +1,7 @@
 #include "GameFeatures/GBFGameFeatureAction_AddWidget.h"
 
 #include "Engine/GBFHUD.h"
+#include "Misc/DataValidation.h"
```
Comment (TheEmidee) [inline Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddWidget.cpp]: <>

File: Source/GameBaseFramework/Private/GameFramework/Phases/GBFGamePhaseAbility.cpp [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 #include "GameFramework/GBFWorldSettings.h"
 #include "GameFramework/Phases/GBFGamePhaseSubsystem.h"
+#include "Misc/DataValidation.h"
```
Comment (TheEmidee) [inline Source/GameBaseFramework/Private/GameFramework/Phases/GBFGamePhaseAbility.cpp]: <>

**Review by TheEmidee — CHANGES_REQUESTED** [review]

File: Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddInputContextMapping.cpp (hunk 1 above)
Comment (Frickerson) [reply Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddInputContextMapping.cpp]: Done.

File: Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddInputContextMapping.cpp (hunk 2 above)
Comment (Frickerson) [reply Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddInputContextMapping.cpp]: Done.

File: Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddWidget.cpp (hunk 3 above)
Comment (Frickerson) [reply Source/GameBaseFramework/Private/GameFeatures/GBFGameFeatureAction_AddWidget.cpp]: Done.

File: Source/GameBaseFramework/Private/GameFramework/Phases/GBFGamePhaseAbility.cpp (hunk 4 above)
Comment (Frickerson) [reply Source/GameBaseFramework/Private/GameFramework/Phases/GBFGamePhaseAbility.cpp]: Done.

**Review by TheEmidee — APPROVED** [review]

**Pull request merged**